business_requirement_deliverable/models/business.py

```diff
@@ -0,0 +1,445 @@
+                    self.partner_id.property_product_pricelist or False,
+            }
+            self.update(values)
+
+    @api.multi
+    def _compute_dl_total_revenue(self):
+        for r in self:
+            r.dl_total_revenue = sum(dl.price_total for dl in
+                                     r.deliverable_lines)
+
+    @api.multi
+    def _compute_dl_count(self):
+        for r in self:
+            r.dl_count = len(r.deliverable_lines.ids)
+
+    @api.multi
+    def _compute_rl_count(self):
+        for r in self:
+            r.rl_count = len(r.resource_lines.ids)
+
+    @api.multi
+    def open_deliverable_line(self):
+        for self in self:
+            domain = [('business_requirement_id', '=', self.id)]
+            br_id = 0
+            if self.state in ('draft', 'confirmed'):
+                br_id = self.id
+            return {
+                'name': _('Deliverable Lines'),
+                'type': 'ir.actions.act_window',
+                'view_type': 'form',
+                'view_mode': 'tree,form,graph',
+                'res_model': 'business.requirement.deliverable',
+                'target': 'current',
+                'domain': domain,
+                'context': {
+                    'tree_view_ref': 'business_requirement_deliverable.' +
+                    'view_business_requirement_deliverable_tree',
+                    'form_view_ref': 'business_requirement_deliverable.' +
+                    'view_business_requirement_deliverable_form',
+                    'default_business_requirement_id': br_id
+                }}
+
+    @api.multi
+    def open_resource_line(self):
+        for self in self:
+            res_lines = self.env['business.requirement.resource'].\
+                search([('business_requirement_id', '=', self.id)])
+            br_id = 0
+            if self.state in ('draft', 'confirmed'):
+                br_id = self.id
+            return {
+                'name': _('Resource Lines'),
+                'view_type': 'form',
+                'view_mode': 'tree,graph',
+                'res_model': 'business.requirement.resource',
+                'type': 'ir.actions.act_window',
+                'domain': [('id', 'in', res_lines.ids)],
+                'context': {
+                    'tree_view_ref': 'business_requirement_resource.' +
+                    'view_business_requirement_resource_tree',
+                    'default_business_requirement_id': br_id
+                }
+            }
+
+    @api.multi
+    @api.depends('pricelist_id')
+    def _compute_get_currency(self):
+        for br in self:
+            if br.partner_id and br.pricelist_id.currency_id:
+                br.currency_id = br.pricelist_id.currency_id.id
+            else:
+                br.currency_id = self.env.user.company_id.currency_id.id
+
+    @api.multi
+    @api.onchange('partner_id')
+    def partner_id_change(self):
```

Comment:
It should be a simple warning.
We should add a onchange for the pricelist -> warning that the pricelist applies only to new deliverable lines.